geopandas/tools/tests/test_sjoin.py

```diff
@@ -499,3 +518,238 @@ def test_sjoin_inner(self):
             self.cities, countries, how="inner", op="intersects"
         )
         assert cities_with_country.shape == (172, 4)
+
+
+@pytest.mark.skipif(
+    TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.options module to"
+        " test sjoin_nearest"
+    ),
+)
+def test_no_nearest_all():
+    df1 = geopandas.GeoDataFrame({"geometry": []})
+    df2 = geopandas.GeoDataFrame({"geometry": []})
+    with pytest.raises(
+        AttributeError, match="Expected the spatial index to implement `nearest_all`"
+    ):
+        sjoin_nearest(df1, df2)
+
+
+@pytest.mark.skipif(
+    not TEST_NEAREST,
+    reason=(
+        f"PyGEOS >= {MIN_PYGEOS_VERSION_FOR_NEAREST}"
+        " must be installed and activated via the geopandas.compat module to"
+        " test sjoin_nearest"
+    ),
+)
+class TestNearest:
+    @pytest.mark.parametrize("how_kwargs", ({}, {"how": "left"}, {"how": "right"}))
+    def test_allowed_hows(self, how_kwargs):
+        left = geopandas.GeoDataFrame({"geometry": []})
+        right = geopandas.GeoDataFrame({"geometry": []})
+        sjoin_nearest(left, right, **how_kwargs)  # no error
+
+    @pytest.mark.parametrize("how", ("inner", "abcde"))
+    def test_invalid_hows(self, how: str):
+        left = geopandas.GeoDataFrame({"geometry": []})
+        right = geopandas.GeoDataFrame({"geometry": []})
+        with pytest.raises(ValueError, match="`how` was"):
+            sjoin_nearest(left, right, how=how)
+
+    @pytest.mark.parametrize("how", ("left", "right"))
```

Comment:
I suggest spliting these functions into more granular tests, which will help avoid the conditional statements within the test itself.  That would keep a single test more focused and easier to review, e.g.,
* left with empty right using left join => all records from left and no results in cols for right
* left with empty right using right join => no records
* left with multiple in right using max_distance = 1, distance_col="dist", and left join => all records from left and only values in right cols for those that are within distance of 1, and that nothing in `dist` column is greater than 1
...

Reply:
@brendan-ward I took the first two suggestions ~ word for word. The third I split out across 4 tests: 2 (how = left, right) for when max_distance results in an "empty" join and 2 for when max_distance results in "unmatched" geometries